cmd/thanos/compact.go

```
@@ -172,6 +173,8 @@ func runCompact(
 		return err
 	}
 
+	confContentYamlStr, _ := config.ReplaceSecret(confContentYaml)
```

It's not a good idea to ignore errors.